In [26]:
# importing important libraries

import pandas as pd
import numpy as np
import statistics as stats
import scipy.stats as stat

#### 1. Scenario: A company wants to analyze the sales performance of its products in different regions. They have collected the following data:
Region A: [10, 15, 12, 8, 14]
    
Region B: [18, 20, 16, 22, 25]

Calculate the mean sales for each region.

In [9]:
df1 = pd.DataFrame(data = {'Region_A': [10, 15, 12, 8, 14], 'Region_B': [18, 20, 16, 22, 25] })
df1

Region_A  Region_B
0        10        18
1        15        20
2        12        16
3         8        22
4        14        25

In [12]:
df1.Region_A.mean(), df1.Region_B.mean()

(11.8, 20.2)

#### 2. Scenario: A survey is conducted to measure customer satisfaction on a scale of 1 to 5. The data collected is as follows:
[4, 5, 2, 3, 5, 4, 3, 2, 4, 5]

Calculate the mode of the survey responses.

In [17]:
cus_satisfaction = [4, 5, 2, 3, 5, 4, 3, 2, 4, 5]
stats.mode(cus_satisfaction)

4

#### 3. Scenario: A company wants to compare the salaries of two departments. The salary data for Department A and Department B are as follows:
Department A: [5000, 6000, 5500, 7000]

Department B: [4500, 5500, 5800, 6000, 5200]

Calculate the median salary for each department.


In [18]:
department_A = [5000, 6000, 5500, 7000]

department_B = [4500, 5500, 5800, 6000, 5200]

In [19]:
np.median(department_A)

5750.0

In [20]:
np.median(department_B)

5500.0

#### 4. Scenario: A data analyst wants to determine the variability in the daily stock prices of a company. The data collected is as follows:
[25.5, 24.8, 26.1, 25.3, 24.9]

Calculate the range of the stock prices.


In [22]:
stock_price = [25.5, 24.8, 26.1, 25.3, 24.9]
stock_range = max(stock_price) - min(stock_price) 
stock_range 

1.3000000000000007

#### 5. Scenario: A study is conducted to compare the performance of two different teaching methods. The test scores of the students in each group are as follows:
Group A: [85, 90, 92, 88, 91]

Group B: [82, 88, 90, 86, 87]

Perform a t-test to determine if there is a significant difference in the mean scores between the two groups.


In [29]:
group_a = [85, 90, 92, 88, 91]

group_b = [82, 88, 90, 86, 87]

alpha_value = 0.05

**Null** : There is no difference between the groups.

**Alternative** : There is difference between the gropus.

In [27]:
t_test_value , p_value = stat.ttest_ind(group_a, group_b)

In [28]:
print('T-test value: ', t_test_value)
print('P-value: ', p_value)

T-test value:  1.4312528946642733
P-value:  0.19023970239078333


In [30]:
if p_value < alpha_value:
    print('we reject the null hypothesis.')
    
else:
    print('We fail to reject the null hypothesis.')

We fail to reject the null hypothesis.


**Conclusion** : We observe from above result that we fail to reject the null hypothesis. It means there is some significant difference between the two groups.

#### 6. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]

Sales (in thousands): [25, 30, 28, 20, 26]

Calculate the correlation coefficient between advertising expenditure and sales.


In [32]:
df2 = pd.DataFrame(data = {'Advertising_Expenditure': [10, 15, 12, 8, 14], 'Sales': [25, 30, 28, 20, 26]})

In [33]:
df2['Advertising_Expenditure'].corr(df2['Sales'])

0.8757511375750132

**Conclusion**: From above result we can observe that there is a significant positive relationship between adertising expenditure and sales of the company.

#### 7. Scenario: A survey is conducted to measure the heights of a group of people. The data collected is as follows:
[160, 170, 165, 155, 175, 180, 170]

Calculate the standard deviation of the heights.



In [35]:
heights = [160, 170, 165, 155, 175, 180, 170]
np.std(heights)

7.953949089757174

#### 8. Scenario: A company wants to analyze the relationship between employee tenure and job satisfaction. The data collected is as follows:
Employee Tenure (in years): [2, 3, 5, 4, 6, 2, 4]

Job Satisfaction (on a scale of 1 to 10): [7, 8, 6, 9, 5, 7, 6]

Perform a linear regression analysis to predict job satisfaction based on employee tenure.


In [36]:
from sklearn.linear_model import LinearRegression

In [55]:
tenure = [2, 3, 5, 4, 6, 2, 4]

job_Satisfaction = [7, 8, 6, 9, 5, 7, 6]

In [56]:
X = np.array([2, 3, 5, 4, 6, 2, 4]).reshape(-1, 1)
y = job_Satisfaction

In [57]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y , random_state= 42, test_size= 0.2)

In [58]:
lr = LinearRegression()

In [59]:
lr.fit(X_train, y_train)

LinearRegression()

In [60]:
y_pred = lr.predict(X_test)

In [61]:
lr.coef_, lr.intercept_

(array([-0.52272727]), 8.795454545454545)

#### 9. Scenario: A study is conducted to compare the effectiveness of two different medications. The recovery times of the patients in each group are as follows:
Medication A: [10, 12, 14, 11, 13]

Medication B: [15, 17, 16, 14, 18]

Perform an analysis of variance (ANOVA) to determine if there is a significant difference in the mean recovery times between the two medications.


In [65]:
medication_a = [10, 12, 14, 11, 13]

medication_b = [15, 17, 16, 14, 18]

In [63]:
f_test_value, p_value = stat.f_oneway(medication_a, medication_b)

In [64]:
print('f-test value: ', f_test_value)
print('P-value: ', p_value)

f-test value:  16.0
P-value:  0.003949772803445326


In [66]:
if p_value < alpha_value:
    print('We reject the null hypothesis')
else:
    print('We fail to reject the null hypothesis')

We reject the null hypothesis


**Conclusion**: From above results we can conclude that there is a difference in recovery times in these two medications.

#### 10. Scenario: A company wants to analyze customer feedback ratings on a scale of 1 to 10. The data collected is as follows:
[8, 9, 7, 6, 8, 10, 9, 8, 7, 8]

Calculate the 75th percentile of the feedback ratings.


In [71]:
cus_feedback = np.array([8, 9, 7, 6, 8, 10, 9, 8, 7, 8])
np.percentile(cus_feedback, 75)

8.75

#### 11. Scenario: A quality control department wants to test the weight consistency of a product. The weights of a sample of products are as follows:
[10.2, 9.8, 10.0, 10.5, 10.3, 10.1]

Perform a hypothesis test to determine if the mean weight differs significantly from 10 grams.

In [73]:
weights = [10.2, 9.8, 10.0, 10.5, 10.3, 10.1]

In [74]:
t_test_stats, p_value = stat.ttest_1samp(weights, 10)

In [76]:
print('T-test-value:', t_test_stats)
print('p_value:', p_value)

T-test-value: 1.5126584522688367
p_value: 0.19077595151110102


In [77]:
if p_value < alpha_value:
    print('We reject the null hypothesis')
else:
    print('We fail to reject the null hypothesis')

We fail to reject the null hypothesis


**Conclusion**: We conclude that there is no significant difference based on above results.

#### 12. Scenario: A company wants to analyze the click-through rates of two different website designs. The number of clicks for each design is as follows:
Design A: [100, 120, 110, 90, 95]

Design B: [80, 85, 90, 95, 100]

Perform a chi-square test to determine if there is a significant difference in the click-through rates between the two designs.



In [78]:
design_a = [100, 120, 110, 90, 95]

design_b = [80, 85, 90, 95, 100]

In [79]:
observed_data = np.array([design_a, design_b])

In [83]:
chi_test = stat.chi2_contingency(observed_data)

In [84]:
if chi_test.pvalue < alpha_value:
    print('We reject the null hypothesis')
    
else:
    print("We fail to reject the null hypothesis")

We fail to reject the null hypothesis


**Conclusion**: We conclude that there is a difference between the two groups.

#### 13. Scenario: A survey is conducted to measure customer satisfaction with a product on a scale of 1 to 10. The data collected is as follows:
[7, 9, 6, 8, 10, 7, 8, 9, 7, 8]

Calculate the 95% confidence interval for the population mean satisfaction score.



In [87]:
data = [7, 9, 6, 8, 10, 7, 8, 9, 7, 8]

In [92]:
# mean, standard_deviation calculation

data_mean = np.mean(data)
data_std = np.std(data)
data_mean, data_std

(7.9, 1.1357816691600546)

In [95]:
# calcuating standard error

std_error = data_std/np.sqrt(len(data))
std_error

0.3591656999213594

In [98]:
# calcuating test stat value

df = len(data) - 1
alpha_value = 1 - 0.95

# based on above values and using t-test tabel we obtain t-critical value to be
t_test_critical = 2.262

In [101]:
# calcuating margin of error

margin_of_error = t_test_critical * std_error
margin_of_error

0.8124328132221149

In [102]:
confidence_interval = (data_mean - margin_of_error, data_mean + margin_of_error)

In [103]:
confidence_interval

(7.0875671867778856, 8.712432813222115)

**Conclusion**: 95% confidence_interval comes out to be (7.0875671867778856, 8.712432813222115)

#### 14. Scenario: A company wants to analyze the effect of temperature on product performance. The data collected is as follows:
Temperature (in degrees Celsius): [20, 22, 23, 19, 21]

Performance (on a scale of 1 to 10): [8, 7, 9, 6, 8]

Perform a simple linear regression to predict performance based on temperature.


In [105]:
temperature = np.array([20, 22, 23, 19, 21])

performance = np.array([8, 7, 9, 6, 8])

In [108]:
X = temperature.reshape(-1,1)
y = performance

In [109]:
X_train, X_test, y_train, y_test = train_test_split(X,y, random_state= 42, test_size= 0.2)

In [110]:
lr2 = LinearRegression()

In [112]:
lr2.fit(X_train,y_train)

LinearRegression()

In [113]:
y_pred = lr2.predict(X_test)

In [116]:
y_pred

array([8.57142857])

In [117]:
lr2.coef_, lr2.intercept_

(array([0.65714286]), -5.885714285714283)

#### 15. Scenario: A study is conducted to compare the preferences of two groups of participants. The preferences are measured on a Likert scale from 1 to 5. The data collected is as follows:
Group A: [4, 3, 5, 2, 4]

Group B: [3, 2, 4, 3, 3]

Perform a Mann-Whitney U test to determine if there is a significant difference in the median preferences between the two groups.



In [119]:
grp_a = [4, 3, 5, 2, 4]
grp_b = [3, 2, 4, 3, 3]

In [121]:
statistic, p_value = stat.mannwhitneyu(preferences_a, preferences_b)

print("Mann-Whitney U statistic:", statistic)
print("p-value:", p_value)

Mann-Whitney U statistic: 17.0
p-value: 0.380836480306712


In [122]:
if p_value < 0.05:
    print('We reject the null hypothesis')
    
else:
    print('We fail to reject the null hypothesis')

We fail to reject the null hypothesis


#### 16. Scenario: A company wants to analyze the distribution of customer ages. The data collected is as follows:
[25, 30, 35, 40, 45, 50, 55, 60, 65, 70]

Calculate the interquartile range (IQR) of the ages.

In [125]:
cus_ages = [25, 30, 35, 40, 45, 50, 55, 60, 65, 70]

In [126]:
IQR = stat.iqr(cus_ages, interpolation = 'midpoint')  
print(IQR)

20.0


#### 17. Scenario: A study is conducted to compare the performance of three different machine learning algorithms. The accuracy scores for each algorithm are as follows:
Algorithm A: [0.85, 0.80, 0.82, 0.87, 0.83]

Algorithm B: [0.78, 0.82, 0.84, 0.80, 0.79]

Algorithm C: [0.90, 0.88, 0.89, 0.86, 0.87]

Perform a Kruskal-Wallis test to determine if there is a significant difference in the median accuracy scores between the algorithms.


In [136]:
scores_a = [0.85, 0.80, 0.82, 0.87, 0.83]
scores_b = [0.78, 0.82, 0.84, 0.80, 0.79]
scores_c = [0.90, 0.88, 0.89, 0.86, 0.87]

In [138]:
statistic, p_value = stat.kruskal(scores_a, scores_b, scores_c)

print("Kruskal-Wallis statistic:", statistic)
print("p-value:", p_value)

Kruskal-Wallis statistic: 9.696947935368053
p-value: 0.007840333026249539


In [139]:
if p_value < 0.05:
    print('We reject the null hypothesis.')
    
else:
    print('We fail to reject the null hypothesis')

We reject the null hypothesis.


#### 18. Scenario: A company wants to analyze the effect of price on sales. The data collected is as follows:
Price (in dollars): [10, 15, 12, 8, 14]

Sales: [100, 80, 90, 110, 95]

Perform a simple linear regression to predict sales based on price.


In [127]:
price = np.array([10, 15, 12, 8, 14]).reshape(-1, 1)
sales = np.array([100, 80, 90, 110, 95])

In [128]:
X = price
y = sales

In [129]:
X_train, X_test, y_train, y_test = train_test_split(X,y, random_state= 42, test_size= 0.2)

In [130]:
lr3 = LinearRegression()

In [132]:
lr3.fit(X_train, y_train)

LinearRegression()

In [134]:
y_predict = lr3.predict(X_test)
y_predict

array([87.75])

In [135]:
lr3.intercept_, lr3.coef_

(129.0, array([-2.75]))

#### 19. Scenario: A survey is conducted to measure the satisfaction levels of customers with a new product. The data collected is as follows:
[7, 8, 9, 6, 8, 7, 9, 7, 8, 7]

Calculate the standard error of the mean satisfaction score.


In [140]:
data1 = [7, 8, 9, 6, 8, 7, 9, 7, 8, 7]

In [141]:
std_error = np.std(data1)/np.sqrt(len(data1))

In [142]:
std_error

0.28982753492378877

#### 20. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]

Sales (in thousands): [25, 30, 28, 20, 26]

Perform a multiple regression analysis to predict sales based on advertising expenditure.



In [143]:
import statsmodels.api as sm

expenditure = np.array([10, 15, 12, 8, 14])

sales = np.array([25, 30, 28, 20, 26])

In [144]:
expenditure_with_intercept = sm.add_constant(expenditure)

In [145]:
model = sm.OLS(sales, expenditure_with_intercept)

results = model.fit()

In [146]:
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.767
Model:                            OLS   Adj. R-squared:                  0.689
Method:                 Least Squares   F-statistic:                     9.872
Date:                Thu, 13 Jul 2023   Prob (F-statistic):             0.0516
Time:                        17:00:07   Log-Likelihood:                -9.5288
No. Observations:                   5   AIC:                             23.06
Df Residuals:                       3   BIC:                             22.28
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         12.2012      4.429      2.755      0.0

C:\Users\nancy\anaconda3\lib\site-packages\statsmodels\stats\stattools.py:74: ValueWarning: omni_normtest is not valid with less than 8 observations; 5 samples were given.
  warn("omni_normtest is not valid with less than 8 observations; %i "
